stripe/src/main/java/com/stripe/android/paymentsheet/ui/BaseSheetActivity.kt

```diff
@@ -33,6 +35,21 @@ internal abstract class BaseSheetActivity<ResultType> : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         supportFragmentManager.fragmentFactory = PaymentSheetFragmentFactory(eventReporter)
+        supportFragmentManager.addOnBackStackChangedListener {
+            if (supportFragmentManager.backStackEntryCount == 0) {
+                toolbar.showClose()
+            } else {
+                toolbar.showBack()
+            }
+        }
+
+        scrollView.viewTreeObserver.addOnScrollChangedListener {
+            appbar.elevation = if (scrollView.scrollY > 0) {
+                resources.getDimension(R.dimen.stripe_paymentsheet_toolbar_elevation)
+            } else {
+                0f
+            }
+        }
```

Comment:
Is there a way to get this behavior for free now?

Reply:
https://stackoverflow.com/questions/36229911/using-bottomsheetbehavior-with-a-inner-coordinatorlayout